contracts/0.8.25/vaults/OperatorGrid.sol

```diff
@@ -0,0 +1,411 @@
+        if (requestedTier.mintedShares + vaultShares > requestedTier.shareLimit) revert TierLimitExceeded();
+
+        //check if group limit is exceeded
+        Group storage group_ = $.groups[nodeOperator];
+        if (group_.mintedShares + vaultShares > group_.shareLimit) revert GroupLimitExceeded();
+
+        Tier storage currentTier = $.tiers[vaultTier.tierIndex];
+
+        currentTier.mintedShares -= uint96(vaultShares);
+        requestedTier.mintedShares += uint96(vaultShares);
+
+        vaultTier.tierIndex = uint128(requestedTierId);
+        vaultTier.tierIndexRequested = 0;
+
+        VaultHub(LIDO_LOCATOR.vaultHub()).updateConnection(
+            _vault,
+            requestedTier.shareLimit,
+            requestedTier.reserveRatioBP,
+            requestedTier.rebalanceThresholdBP,
+            requestedTier.treasuryFeeBP
+        );
+
+        emit TierChanged(_vault, requestedTierId);
+    }
+
+   // -----------------------------
+   //     MINT / BURN
+   // -----------------------------
+
+    /// @notice Mint shares limit check
+    /// @param vaultAddr address of the vault
+    /// @param amount amount of shares will be minted
+    function onMintedShares(
+        address vaultAddr,
+        uint256 amount
+    ) external {
+        if (msg.sender != LIDO_LOCATOR.vaultHub()) revert NotAuthorized("onMintedShares", msg.sender);
+
+        ERC7201Storage storage $ = _getStorage();
+
+        VaultTier memory vaultTier = $.vaultTier[vaultAddr];
+        uint256 tierId = vaultTier.tierIndex;
+        if (tierId == 0) revert VaultNotExists();
+
+        Tier storage tier = $.tiers[tierId];
+        Group storage group_ = $.groups[tier.operator];
+
+        uint96 amount_ = uint96(amount);
+
+        if (tier.mintedShares + amount_ > tier.shareLimit) revert TierLimitExceeded();
+        if (group_.mintedShares + amount_ > group_.shareLimit) revert GroupLimitExceeded();
+
+        tier.mintedShares += amount_;
+        group_.mintedShares += amount_;
+
+        emit SharesLimitChanged(tier.operator, tierId, vaultAddr, tier.mintedShares, group_.mintedShares);
+    }
+
+    /// @notice Burn shares limit check
+    /// @param vaultAddr address of the vault
+    /// @param amount amount of shares to burn
+    function onBurnedShares(
+        address vaultAddr,
+        uint256 amount
+    ) external {
+        if (msg.sender != LIDO_LOCATOR.vaultHub()) revert NotAuthorized("burnShares", msg.sender);
+
+        ERC7201Storage storage $ = _getStorage();
+
+        VaultTier memory vaultTier = $.vaultTier[vaultAddr];
+        uint256 tierId = vaultTier.tierIndex;
+        if (tierId == 0) revert VaultNotExists();
+
+        Tier storage tier = $.tiers[tierId];
+        Group storage group_ = $.groups[tier.operator];
+
+        uint96 amount_ = uint96(amount);
+
+        if (group_.mintedShares < amount_) revert GroupMintedSharesUnderflow();
+        if (tier.mintedShares < amount_) revert TierMintedSharesUnderflow();
```

Comment:
Maybe remove this check at all. It will fail anyway, but save us some gas.

Reply:
removed, already checked in vaultHub